server/image_pull.go

```diff
@@ -70,29 +70,25 @@ func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (resp
 			}
 		}
 
-		var canPull bool
-		canPull, err = s.StorageImageServer().CanPull(img, options)
-		if err != nil && !canPull {
-			logrus.Debugf("error checking image %s: %v", img, err)
+		tmpImg, err := s.StorageImageServer().PrepareImage(s.ImageContext(), img, options)
```

Comment:
BTW, in case this code stays long-term, it would be worth adding a comment here, explaining that `PrepareImage` also serves as “fail if the image is not accessible on the registry” check, and therefore needs to happen even if `storedImage == nil`, even though the `storedImage` check is probably faster.